src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/TerminalTestReporter.cs

```diff
@@ -647,7 +647,16 @@ private static void AppendAssemblyLinkTargetFrameworkAndArchitecture(ITerminal t
     internal /* for testing */ static void AppendStackFrame(ITerminal terminal, string stackTraceLine)
     {
         terminal.Append(DoubleIndentation);
-        Match match = GetFrameRegex().Match(stackTraceLine);
+        Match match;
+        try
+        {
+            match = GetFrameRegex().Match(stackTraceLine);
+        }
+        catch (RegexMatchTimeoutException ex)
+        {
+            throw new RegexMatchTimeoutException(string.Format(CultureInfo.CurrentCulture, PlatformResources.TimeoutInRegexStackTraceLineParsing, stackTraceLine), ex);
```

Comment:
Do we want to preserve the original stack trace?

Maybe with `ExceptionDispatchInfo.Capture(ex).Throw(new RegexMatchTimeoutException(...));`

Reply:
ok added